File: Leetcode July Challenge/C++/Reshape the Matrix.cpp

// Approach - 1
class Solution {
public:
    vector<vector<int>> matrixReshape(vector<vector<int>>& mat, int r, int c) {
        int rows = mat.size();
        int cols = mat[0].size();
        if(rows*cols < r*c || rows*cols > r*c){
            return mat;
        }
        
        vector<int> me;
        for(int i = 0; i < rows; i++){
            for(int j = 0; j < cols; j++){
                me.push_back(mat[i][j]);
            }
        }
        
        int k = 0;
        vector<vector<int>> new_matrix(r, vector<int>(c, 0));
        for(int i = 0; i < r; i++){
            for(int j = 0; j < c; j++){
                new_matrix[i][j] = me[k];
                k++;
            }
        }
        
        return new_matrix;
        
    }
};


// Approach - 2
class Solution {
public:
    vector<vector<int>> matrixReshape(vector<vector<int>>& nums, int r, int c) {
        int m = nums.size(), n = nums[0].size(), o = m * n;
        if (r * c != o) return nums;
        vector<vector<int>> res(r, vector<int>(c, 0));
        for (int i = 0; i < o; i++) res[i / c][i % c] = nums[i / n][i % n];
        return res;
    }
};

File: Leetcode July Challenge/Python/Reshape the Matrix.py


# Approach -1
class Solution:
    def matrixReshape(self, mat: List[List[int]], r: int, c: int) -> List[List[int]]:
        rows = len(mat)
        cols = len(mat[0])
        if rows*cols < r*c or rows*cols > r*c:
            return mat
        
        me = []
        for i in range(rows):
            for j in range(cols):
                me.append(mat[i][j])
        
        new_matrix = [[0 for j in range(c)] for i in range(r)]
        
        k = 0
        for i in range(r):
            for j in range(c):
                new_matrix[i][j] = me[k]
                k+=1
        
        return new_matrix
    
    
        
        
